test/unit/state_count_reporter_test.rb

```diff
@@ -14,8 +14,8 @@ class StateCountReporterTest < ActiveSupport::TestCase
   end
 
   should "report a count for each state" do
-    statsd_mock = mock("statsd") do
-      stubs(:batch).yields(self)
+    statsd_mock = mock("statsd")
+    statsd_mock.stubs(:batch).yields(self).tap do
       expects(:gauge).with("state.draft", 12)
       expects(:gauge).with("state.published", 15)
     end
```

Comment:
> Btw, looks like you could've just added `.tap` to the end of `mock("statsd")` and it would be correct without changing anything else. Like:
> 
> ```diff
> - statsd_mock = mock("statsd") do 
> + statsd_mock = mock("statsd").tap do 
> ```

That doesn't work, I get this error `expected exactly once, invoked never: #<StateCountReporterTest:0x563fe20b51a8>.gauge("state.draft", 12)`

This was a bit weird and I spent ages trying to work around it. This is the original minitest warning I got `Use Object#tap or define stubs/expectations with an explicit receiver instead.`